include/io.h

```diff
@@ -19,15 +20,13 @@
 using json = nlohmann::json;
 
 //! \brief Input/Output handler
-//! \tparam Tdim dimension
-template <unsigned Tdim>
 class IO {
  public:
   //! Constructor with json input file
   //! Get mesh_filename and output_directory
   //! \param[in] input directory
   //! \param[in] json input file name
-  explicit IO(const std::string& file_directory, const std::string& json_file);
+  explicit IO(const int& argc, const char**& argv);
```

Comment:
You don't need explicit when you have more than one argument. and it's `int` and `char`, so drop `const` and `reference`